source/exe/hot_restart.cc

```diff
@@ -173,7 +173,7 @@ int HotRestartImpl::duplicateParentListenSocket(const std::string& address) {
   if (address.length() >= sizeof(rpc.address_)) {
     return -1;
   }
-  StringUtil::strlcpy(rpc.address_, address.c_str(), address.length());
+  StringUtil::strlcpy(rpc.address_, address.c_str(), address.length() + 1);
```

Comment:
The last param is the max length of the target buffer, which is why it didn't work before.